firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/internal/ReleaseIdentificationUtils.java

```diff
@@ -40,4 +50,70 @@ public static String extractInternalAppSharingArtifactId(@NonNull Context appCon
       return null;
     }
   }
+
+  @Nullable
+  public static String calculateApkInternalCodeHash(@NonNull File file) {
+    Log.v(TAG, String.format("Calculating release id for %s", file.getPath()));
+    Log.v(TAG, String.format("File size: %d", file.length()));
+
+    long start = System.currentTimeMillis();
+    long entries = 0;
+    String zipFingerprint = null;
+    try {
+      MessageDigest digest = MessageDigest.getInstance("SHA-256");
+      ArrayList<Byte> checksums = new ArrayList<>();
+      ZipFile zis = new ZipFile(file);
```

Comment:
Could we add a comment explaining why we hash the zip file?

Reply:
+1 I think high-level comments of what we do in this class or this method would be really useful for the future